### ISSUE TITLE
How to get the name of the icon component in Figma within figma.connect method?

### ISSUE THREAD

**Issue by stacysun-cb** [figma/code-connect#145]
I have connected all the icons assets in Figma. I used scripts to generate code connect files and added `name` props using the component name like below. 
<img width="701" alt="Screenshot 2024-08-28 at 12 58 17 PM" src="https://github.com/user-attachments/assets/69f54ad0-dadd-4146-a83c-bd9cf2af13d7">

Now on IconButton component, it shows inconsistent behavior for `name` prop. Some icon renders code and some shows in instances. I was using `name: figma.instance('icon')` for the `name` prop. 
<img width="696" alt="Screenshot 2024-08-28 at 1 03 41 PM" src="https://github.com/user-attachments/assets/bf75e6a0-fd2d-46aa-ab7b-4b5807c62459">
<img width="693" alt="Screenshot 2024-08-28 at 1 03 55 PM" src="https://github.com/user-attachments/assets/004dcd31-b958-42b2-8cbe-a5133253f19f">

My questions:
- Despite the inconsistency, how to get the `name` prop on icon instances? The `name` props is not a design prop, but generated through script. We tried to add an `name` prop in Figma for each icon, but it seems there's no proper variable type to use.

Thanks!

Please provide:
Code Connect CLI version: 1.0.6
Operating system: macOS

**Comment by karlpetersson** [comment]
Hi @stacysun-cb, regarding extracting `name` from an icon, this is not currently possible but something we're working on adding - I don't have any details to share just yet but hope to support this soon.

As for the issue you're having: Do you see any errors when publishing or otherwise? Usually, we will show pills for instances that either (1) couldn't render in the parent component or (2) doesn't have code connect attached to them. In this case, could this specific icon have been missed and not be published yet?

It would be helpful if you could provide the `figma.connect` call for this specific icon where it doesn't work.

**Comment by stacysun-cb** [comment]
@karlpetersson Thanks for following up! We decided to hardcode for `name` prop of the icon. Looking forward to hearing the updates on supporting this feature.

**Comment by ericandrewscott** [comment]
@karlpetersson We've run into this as well. In our design system, to avoid the temptation for product designers to change the color or size of icons that are used in other components (like button, input, etc) we have chosen to create a separate glyph component that gets used in our other design components. This glyph is different than our actual icon component.

However, this glyph does not exist in actual React code (or native platforms). To get around the issue @stacysun-cb is describing, is there a way to generate "code" for the glyphs that quite literally just returns a string, eg `info`, `close`, `reply`, `child-friendly`, etc, so that anywhere we use those glyphs in a component, we can tell Code Connect that we care about the glyph children, and it returns the string. 

Is there a way to do that? We're envisioning that instead of it returning an actual React component, it would return the string that we can use as a prop for our other components.

**Comment by carrybit** [comment]
W also have this need. We have configured code connect for an icon component in Figma, as well as a component (Box in the example) which uses the said icon. 

```
<Box
  icon={<Icon type="Smiley" scale={0.5} />}
/>
```

We want the size of the icon when displayed inside Box to be decided by the box itself. 

By having the icon set as an instance, nothing is guarding the size from being set to whatever size. By instead passing in "Smiley" as a prop, Box can set the size of the icon itself.

```
<Box
  title="Link text"
  description="Description of content"
  destination="internal"
  icon="Smiley" />}
/>
```

```
const Box = (icon: IconType) => {
	<View>
		...

                {icon && <Icon type={icon} scale={0.5} />}
	</View>
	
}
```


An alternative approach to this is to configure code connect for the Smiley illustration as: 

```
<Smiley />
```

And then pass in the component type of Smiley as a property to Box, i.e. 

```
<Box
	title="Link text"
	description="Description of content"
	destination="internal"
	icon={Smiley} />}
/>
```

Box can then set size props when instantiating the icon:

```
interface Props
{
	icon: ComponentType<{scale: number}>
}

const Box = ({icon: Icon}: Props) => {
	<View>
		...

                {Icon && <Icon scale={0.5} />}
	</View>
	
}
```

**Comment by slees-figma** [comment]
Hi @stacysun-cb @carrybit @ericandrewscott, we've just released v1.2.0 which includes some changes to better support icons. The easiest way to achieve this is by returning a string in your icon Code Connect:
```typescript

figma.connect("icon", {
  example: () => "my-icon"
})

// `icon` will render a string instead of a JSX element
figma.connect("parent", {
  props: {
    icon: figma.instance<string>("IconInstance")
  },
  example: ({ icon }) => <Box icon={icon} />
})
```
If your Icon code connect has additional props, we now also support a `getProps` helper to extract these:
```typescript

figma.connect("icon", {
  props: {
    iconId: "my-icon",
  },
  example: ({ iconId }) => <Icon id={iconId} />
})

figma.connect("parent", {
  props: {
    iconProps: figma.instance("IconInstance").getProps<{ iconId: string }>()
  },
  example: ({ iconProps }) => <Box icon={iconProps.iconId} />
})
```

See the [docs](https://github.com/figma/code-connect/blob/main/docs/react.md#icons) for more info, and let us know if you have any questions!

**Comment by narwold** [comment]
We have no interest in CodeConnecting hundreds of icons. There needs to be a way to read the name of the component.
